### PR TITLE
Add React Diagram dependency

### PR DESCRIPTION
**Pull request by ncarroll-mdsol**
Adds the react diagram dependency and gets a very simple graph displayed on the page.

### CODE REVIEW

File: components/WorkflowBuilderPage.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+} from '@projectstorm/react-diagrams';
+
+import { CanvasWidget } from '@projectstorm/react-canvas-core';
+
+const WorkflowBuilderPage = () => {
+
+  //const [engine] = useState([]);
+
+    //1) setup the diagram engine
+    var engine = createEngine();
```
Comment (kaspatel-mdsol) [inline components/WorkflowBuilderPage.js:18]: For any variables, use `useRef` hook as follows.
```
const engine = useRef(createEngine())
const model = useRef(new DiagramModel())
...
model.current.addAll(...)
engine.current.setModel(model.current)
...
<CanvasWidget engine={engine.current} />
```
Again, this is something that will break anything, but it's what react recommends, like if you need a reference to DOMNode or a library mixin like in this case.

File: package.json [hunk 2]
```diff
@@ -12,10 +12,20 @@
     "start": "next start"
   },
   "dependencies": {
+    "@emotion/core": "^10.0.28",
+    "@emotion/styled": "^10.0.27",
```
Comment (kaspatel-mdsol) [inline package.json:16]: We don't need this.

File: package.json [hunk 3]
```diff
@@ -12,10 +12,20 @@
...
+    "lodash": "^4.17.15",
+    "mathjs": "^6.6.4",
     "next": "^9.3.5",
+    "pathfinding": "^0.4.18",
+    "paths-js": "^0.4.10",
     "react": "^16.13.1",
     "react-beautiful-dnd": "^13.0.0",
-    "react-dom": "^16.13.1"
+    "react-dom": "^16.13.1",
+    "resize-observer-polyfill": "^1.5.1"
```
Comment (kaspatel-mdsol) [inline package.json:28]: Same here.

File: styles/WorkflowBuilderPage.module.css [hunk 4]
```diff
@@ -0,0 +1,9 @@
+.workflowbuilder-container {
+  display: grid;
+  grid-template-columns: 3fr 1fr;
```
Comment (kaspatel-mdsol) [inline styles/WorkflowBuilderPage.module.css:3]: Also, we want the whole screen for this, so the following should suffice. Currently, if you drag the node to extreme right, it is hidden.
```
grid-template-columns: 1fr
```

File: styles/WorkflowBuilderPage.module.css [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+  height: 100%;
+}
+
+.workflowbuilder-container aside {
```
Comment (kaspatel-mdsol) [inline styles/WorkflowBuilderPage.module.css:7]: Also, we don't need this rn.

File: components/WorkflowBuilderPage.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+    //2) setup the diagram model
+    var model = new DiagramModel();
+
+    //3-A) create a default node
+    var node1 = new DefaultNodeModel({
```
Comment (kaspatel-mdsol) [inline components/WorkflowBuilderPage.js:24]: Also, in JS now we only use `const` and `let`. Not using `var` fixes any hoisting issues that give JS a bad name.

As these variables never change, we use `const`

File: package.json (hunk 2 above)
Comment (ncarroll-mdsol) [reply package.json:16]: Without this and the `resize-observer-polyfill` dependency below, I am unable to build and see the following when installing dependencies:

> warning "@projectstorm/react-diagrams > @projectstorm/react-diagrams-core@6.0.2" has unmet peer dependency "resize-observer-polyfill@^1.5.1".
warning "@projectstorm/react-diagrams > @projectstorm/react-diagrams-defaults@6.0.2" has unmet peer dependency "@emotion/styled@^10.*".

File: components/WorkflowBuilderPage.js (hunk 1 above)
Comment (ncarroll-mdsol) [reply components/WorkflowBuilderPage.js]: It's kinda funny that React now has a way to replace any variable. I don't quite get `useRef`even after reading the documentation. It really sounds like it's only useful if I'm binding the variable to the DOM? 

> The only difference between useRef() and creating a {current: ...} object yourself is that useRef will give you the same ref object on every render.

But in any case, happy to use it if it's the official recommendation.

File: styles/WorkflowBuilderPage.module.css [hunk 7]
```diff
@@ -1,9 +1,5 @@
...
   display: grid;
-  grid-template-columns: 3fr 1fr;
+  grid-template-columns: 1fr;
   height: 100%;
-}
-
-.workflowbuilder-container aside {
-  background-color: antiquewhite;
-}
+}
```
Comment (kaspatel-mdsol) [inline styles/WorkflowBuilderPage.module.css:5]: New line here

**Pull request merged**